granary/reddit.py

```diff
@@ -0,0 +1,215 @@
+          typ: string to denote whether to get submission or comment content
+
+        Returns:
+          an ActivityStreams object dict, ready to be JSON-encoded
+            """
+        obj = {}
+
+        # always prefer id_str over id to avoid any chance of integer overflow.
+        # usually shouldn't matter in Python, but still.
+        id = thng.id
+        if not id:
+            return {}
+
+        created_at = thng.created_utc
+        try:
+            published = self.timestamp_to_iso8601(created_at)
+        except ValueError:
+            # this is probably already ISO 8601, likely from the archive export.
+            # https://help.twitter.com/en/managing-your-account/how-to-download-your-twitter-archive
+            # https://chat.indieweb.org/dev/2018-03-30#t1522442860737900
+            published = created_at
+
+        obj = {
+        'id': self.tag_uri(id),
+        'objectType': 'note',
+        'published': published,
+        'to': [],
+        }
+
+        user = thng.author
+        if user:
+            obj['author'] = self.redditor_to_actor(user)
+            username = obj['author'].get('username')
+
+        obj['to'].append({
+                          'objectType': 'group',
+                          'alias': '@public',
+                          })
+
+        obj['url'] = self.BASE_URL+thng.permalink
+
+        if typ=='submission':
+            obj['content'] = thng.title
+        elif typ=='comment':
+            obj['content'] = thng.body
+
+        return self.postprocess_object(obj)
+
+
+    def praw_to_activity(self, thng, typ):
+        """Converts a praw submission or comment to an activity.
+        https://praw.readthedocs.io/en/latest/code_overview/models/submission.html
+        https://praw.readthedocs.io/en/latest/code_overview/models/comment.html
+
+        Args:
+          thng: a praw object, Sumbission or Comment
+          typ: string to denote whether to get submission or comment content
+
+        Returns:
+          an ActivityStreams activity dict, ready to be JSON-encoded
+        """
+        obj = self.praw_to_object(thng, typ)
+        actor = obj['author']
+
+        activity = {
+          'verb': 'post',
+          'id': thng.id,
+          'url': self.BASE_URL+thng.permalink,
+          'actor': actor,
+          'object': obj
+          }
+
+        return self.postprocess_activity(activity)
+
+    def fetch_replies(self, r, activities):
+        """Fetches and injects reddit comments into a list of activities, in place.
+
+        limitations: Only includes top level comments
+        Args:
+          r: praw api object for querying submissions in activities
+          activities: list of activity dicts
+
+        Returns:
+          same activities list
+        """
+        for activity in activities:
+            subm = r.submission(id=activity['id'])
+
+            # for v0 we will use just the top level comments because threading is hard
```

Comment:
sgtm. consider mentioning this in the docstring. can updated it later if we add descendants.

Reply:
i think it was included in docstring already "limitations: Only includes top level comments"